src/ai/ai_help_structs.cc

```diff
@@ -1041,15 +1053,22 @@ void PlayersStrengths::remove_stat(const Widelands::PlayerNumber pn) {
 // After statistics for team members are updated, this calculation is needed
 void PlayersStrengths::recalculate_team_power() {
 	team_powers.clear();
+	team_members.clear();
+	active_players_ = 0U;
 	for (auto& item : all_stats) {
 		if (item.second.team_number > 0) {  // is a member of a team
 			if (team_powers.count(item.second.team_number) > 0) {
 				team_powers[item.second.team_number] += item.second.players_power;
 			} else {
 				team_powers[item.second.team_number] = item.second.players_power;
 			}
+			++team_members[item.second.team_number];
+		}
+		if (!item.second.defeated) {
+			++active_players_;
 		}
 	}
```

Comment:
```suggestion
	for (auto& item : all_stats) {
		if (item.second.defeated) {
			continue;
		}
		++active_players_;
		if (item.second.team_number > 0) {  // is a member of a team
			if (team_powers.count(item.second.team_number) > 0) {
				team_powers[item.second.team_number] += item.second.players_power;
			} else {
				team_powers[item.second.team_number] = item.second.players_power;
			}
			++team_members[item.second.team_number];
		}
	}
```

Reply:
This is actually just moving the check for defeated to the beginning of the loop and skip everything for defeated players. I think github shows it as a big diff because previously unchanged lines are affected too.